### PR TITLE
add sig v4 message handler

### PR DESCRIPTION
**Pull request by jorros**

### CODE REVIEW

File: src/SigV4MessageHandler/Crypto.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+using System.Security.Cryptography;
+
+namespace InsightArchitectures.Utilities;
+
+#pragma warning disable SA1600
```
Comment (Kralizek) [inline src/SigV4MessageHandler/Crypto.cs:5]: enable it again

File: src/SigV4MessageHandler/SigV4MessageHandler.cs [hunk 2]
```diff
@@ -0,0 +1,232 @@
+using System.Globalization;
```
Comment (Kralizek) [inline src/SigV4MessageHandler/SigV4MessageHandler.cs:1]: we might need to include the Amazon header to each of the files we copied from the SDK.

**Review by Kralizek — CHANGES_REQUESTED** [review]
It looks good. I would rename the package into something that makes the intent clearer. Maybe something like `InsightArchitectures.Utilities.Http.AWSSignature` (let's take this tomorrow)

Check the license from AWS regarding redistributing their code. They should be fine with it, granted we explicitly say where the files/classes come from.

Finally, add a package description and tags to the project file.

File: src/SigV4MessageHandler/Crypto.cs (hunk 1 above)
Comment (jorros) [reply src/SigV4MessageHandler/Crypto.cs:5]: Hmm, this is an internal class. Does it need documentation?

Comment (Kralizek) [reply src/SigV4MessageHandler/Crypto.cs]: No, i meant at the end of the file.

File: src/AWSSignatureV4/AWSSignatureV4.csproj [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="AWSSDK.Core" Version="3.3.0" />
+    <PackageReference Include="System.Net.Http" Version="4.3.4" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <PackageDescription>Contains a message handler for HttpClient, that allows calling IAM secured endpoints using Signature 4 signing.</PackageDescription>
+    <PackageTags>dotnet;utilities;aws;signature;v4;iam;httpclient</PackageTags>
```
Comment (Kralizek) [inline src/AWSSignatureV4/AWSSignatureV4.csproj:18]:
```suggestion
    <PackageTags>dotnet;utilities;aws;aws-signature-v4;aws-iam;httpclient</PackageTags>
```

**Review by Kralizek — CHANGES_REQUESTED** [review]
A small change for the tags.

Also check if you need to override anything from the `/src/Directory.Build.props` file

**Pull request merged**